tests/user/views/registration.py

```diff
@@ -0,0 +1,200 @@
+from bs4 import BeautifulSoup
+from django.contrib.auth.models import User
+from django.core import mail
+
+from tests.user.views.wizard_helper import WizardViewTestCase
+
+
```

Comment:
Maybe add tests to make sure terms and conditions are agreed to, and that the newsletter preferences is set correctly, and also correctly updated on sendgrid, as those two things are quite important?

Reply:
from what I can tell, the check is done client side via javascript but I'm not sure how to confirm this. mind looking into it?

Reply:
afaik no js is involved, all validation is done in python, and it renders back the view if it is invalid no? Or do you mean the required tag? Cause that is just plain html, but a test should be made anyway, cause we want to validate correct behaviour for this.